[#70194] Primerize Admin Backlogs Settings

myabc opened the pull request:
# Ticket

https://community.openproject.org/work_packages/70194

# What are you trying to accomplish?

Primerizes Administration > Backlogs page

## Screenshots

<img width="749" height="497" alt="Screenshot 2026-01-06 at 22 55 30" src="https://github.com/user-attachments/assets/3bcef25c-643f-42ee-8b65-eb1000ce2cbb" />

<img width="991" height="665" alt="image" src="https://github.com/user-attachments/assets/9d96d870-baca-4b02-8098-984f56c96242" />

<img width="1227" height="698" alt="Screenshot 2026-01-06 at 21 32 25" src="https://github.com/user-attachments/assets/9030499f-7abd-45df-bbd3-4045b322b65e" />


# What approach did you choose and why?
<!-- This section is a place for you to describe your thought process in making these changes.
     List any tradeoffs you made to take on or pay down tech debt.
     Describe any alternative approaches you considered and why you discarded them. -->

# Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

myabc committed:
[#70194] Primerize Backlogs Admin Settings

https://community.openproject.org/work_packages/70194

myabc committed:
Implement select panel form control/input

Makes `SelectPanel` easily usable via the Forms DSL. Based on upstream
`ActionMenuInput` implementation.

myabc committed:
Add feature spec for Backlogs Admin Settings

myabc committed:
Use Form model to DRY up Form, tidy validations

myabc commented on frontend/src/stimulus/controllers/dynamic/admin/backlogs-settings.controller.ts:
```diff
@@ -0,0 +1,131 @@
+
+  declare readonly storyTypesTarget:SelectPanelElement;
+  declare readonly taskTypeTarget:SelectPanelElement;
+  declare readonly hasStoryTypesTarget:boolean;
+  declare readonly hasTaskTypeTarget:boolean;
```
this doesn't conform to our eslint settings.

myabc commented on modules/backlogs/app/views/backlogs_settings/show.html.erb:
```diff
@@ -1,4 +1,4 @@
-<%#-- copyright
+<%# -- copyright
```
Herb changes formatting of ERb files on save - and it's not worth configuring an exception.

myabc committed:
Removes list padding from all ActionLists

HDinger reviewed (approved):
LGTM 👍 

As discussed, I'd like to get a review from product people as well before we merge this.

HDinger commented:
As discussed yesterday, there are two issues worth fixing:

* Hiding the disabled items from the options
* Show the correct label when de-selecting all items

myabc committed:
Add an Apply button to multi-select select panel

Also renames submit button from "Apply" to "Save" to avoid confusion and
be more consistent with other Admin forms.

myabc committed:
Make mutual exclusivity in type selection clearer

- Fixes styling of disabled select panel items (adds an override that
  should eventually be removed in favor of removing `!important` from
  highlighting styles).
- Adds accompanying field captions.

myabc commented:
> As discussed yesterday, there are two issues worth fixing:
> 
> * Hiding the disabled items from the options
> * Show the correct label when de-selecting all items

@HDinger ok, this is what I came up with:

<img width="354" height="584" alt="image" src="https://github.com/user-attachments/assets/fbe4b55e-e778-420c-97cb-fb520dd89d45" />

I was able to fix the disabled styling by means of an override. The highlighting styles makes this difficult because of the use of `!important` - I may add a Code Maintenance ticket to see if we can remove this at some point.

I considered adding tooltips, but showing different tooltips depending on the item's disabled status is not something select panel supports out of the box. Finding an elegant solution would involve more than a couple hour's work. 

In the end I added additional captions, which I think looks OK:

<img width="526" height="520" alt="image" src="https://github.com/user-attachments/assets/14a7fd76-6df9-4395-8eec-1379a37d1cfc" />

I will go ahead and merge since this is "good enough" for now (and much better than before).

myabc commented:
@HDinger FYI - as with other Settings page we don't show a required indicator. Correct me if I'm wrong, but I don't think there is such a thing as an optional Setting per sé - just Settings with specified defaults and empty defaults.

myabc commented on frontend/src/global_styles/primer/_overrides.sass:
```diff
@@ -152,3 +148,7 @@ ul.SegmentedControl,
       overflow: visible
       text-overflow: clip
+
+.ActionListContent[aria-disabled="true"]
+  .ActionListItem-label[class^="__hl_"], .ActionListItem-label[class*=" __hl_"]
+    color: var(--control-fgColor-disabled) !important
```
FUGLY code
